oeps/oep-0021-proc-deprecation.rst

```diff
@@ -0,0 +1,310 @@
+|                 | Nimisha Asthagiri <nimisha@edx.org>                    |
++-----------------+--------------------------------------------------------+
+| Arbiter         | Diana Huang <dhuang@edx.org>                           |
++-----------------+--------------------------------------------------------+
+| Status          | Under Review                                           |
++-----------------+--------------------------------------------------------+
+| Type            | Process                                                |
++-----------------+--------------------------------------------------------+
+| Created         | 2018-05-18                                             |
++-----------------+--------------------------------------------------------+
+| Review Period   | 2018-06-06 - 2018-06-20, `open-edx-proposals#63`_      |
++-----------------+--------------------------------------------------------+
+| Resolution      |                                                        |
++-----------------+--------------------------------------------------------+
+| References      |                                                        |
++-----------------+--------------------------------------------------------+
+
+.. _open-edx-proposals#63: https://github.com/edx/open-edx-proposals/pull/63
+
+Abstract
+========
+
+Open edX needs a formal process on how to deprecate and remove unused or
+unneeded technology.
+
+This document details that process.
+
+Motivation
+==========
+
+There currently is no formal process on how to identify and deprecate
+technology that can be removed from the Open edX platform. As a result, the
+state of technologies that can be removed or are already partially
+removed is captured in a non-standardized set of formats. This document outlines
+common language and a process to use when making the decision on what to
+deprecate, when it should be done, and the status of execution. The result of
+following this process should be alignment and the right degree of involvement
+from all stakeholders.
+
+Specification
+==============
+
+When a team or individual believes that any technology on the Open edX
+platform should no longer be used or maintained, the following steps should be
+followed.
+
+Criteria and identification
+---------------------------
+
+The list of technology that should be considered for depreciation and
+removal from the edX platform includes but is not limited to:
+
+- REST APIs
+
+- Apps
+
+- Learner or partner facing features (user interfaces and associated backend)
+
+- Backend features - infrastructure and integrations
+
+- Feature toggles (multiple code paths)
+
+A general rule to follow is that if it’s code that's used by any Open edX
+platform feature then the associated deprecation/removal should be handled by
+the process outlined in this OEP.
+
+Once the technology to be deprecated is identified, the originator
+should be prepared to communicate the following:
```

Comment:
Who can originate a deprecation? 

If you want community members to originate deprecations, it may be difficult for them to do the necessary JIRA tasks.

Reply:
@pdpinch Thanks for bringing this up. The intent is to have community members able to originate deprecation as well, so I'll make that clearer.

As part of having edX IT setup our Deprecation Kanban board, I'll work with them to make sure that community members are able to do what this OEP asks. If it turns out that's not possible, then the OEP will be updated accordingly.